Source/WebCore/platform/mediastream/WebRTCProvider.cpp

```diff
@@ -198,6 +203,131 @@ bool WebRTCProvider::isH264EncoderSmooth(const VideoConfiguration&)
+            parameters.generalTierFlag = 0;
+        else if (tierFlag == "1"_s)
+            parameters.generalTierFlag = 1;
+        else
+            return { };
+
+        auto levelId = contentType.parameter("level-id"_s);
+        if (!levelId.isEmpty()) {
+            auto result = parseInteger<uint16_t>(levelId);
+            if (!result)
+                return { };
+            parameters.generalLevelIDC = *result;
+        } else
+            parameters.generalLevelIDC = 93;
+
+        return makeString("video/mp4;codecs="_s, createHEVCCodecParametersString(parameters));
+    }
+
+    if (equalLettersIgnoringASCIICase(containerType, "video/vp9"_s)) {
+        // https://www.rfc-editor.org/rfc/rfc9628.html#payloadFormatParameters
+        VPCodecConfigurationRecord parameters;
+        parameters.codecName = "vp09"_s;
+
+        auto profileId = contentType.parameter("profile-id"_s);
+        if (!profileId.isEmpty()) {
+            auto result = parseInteger<uint8_t>(profileId);
+            if (!result || *result > 3)
+                return { };
+            parameters.profile = *result;
+        }
+
+        // We use level 5 as the default.
+        parameters.level = VPConfigurationLevel::Level_5;
+
+        parameters.bitDepth = parameters.profile < 2 ? 8 : 10;
+
+        return makeString("video/mp4;codecs="_s, createVPCodecParametersString(parameters));
+    }
+
+    if (equalLettersIgnoringASCIICase(containerType, "video/av1"_s)) {
+        // https://www.iana.org/assignments/media-types/video/AV1
+        AV1CodecConfigurationRecord parameters;
+        parameters.codecName = "av01"_s;
+
+        auto profile = contentType.parameter("profile"_s);
+        if (!profile.isEmpty()) {
+            auto result = parseEnumFromStringView<AV1ConfigurationProfile>(profile);
+            if (!result)
+                return { };
+            parameters.profile = *result;
+        }
+
+        auto levelIdx = contentType.parameter("level-idx"_s);
+        if (!levelIdx.isEmpty()) {
+            auto result = parseEnumFromStringView<AV1ConfigurationLevel>(levelIdx);
+            if (!result)
+                return { };
+            parameters.level = *result;
+        } else
+            parameters.level = AV1ConfigurationLevel::Level_3_1;
+
+        auto tier = contentType.parameter("tier"_s);
+        if (!tier.isEmpty()) {
+            if (tier == "M"_s)
+                parameters.tier = AV1ConfigurationTier::Main;
+            else if (tier == "H"_s)
+                parameters.tier = AV1ConfigurationTier::High;
+            else
+                return { };
+        }
+
+        parameters.bitDepth = parameters.profile < AV1ConfigurationProfile::Professional ? 8 : 10;
+
+        return makeString("video/mp4;codecs="_s, createAV1CodecParametersString(parameters));
+    }
+
+    if (equalLettersIgnoringASCIICase(containerType, "video/vp8"_s))
+        return "video/webm;codecs=vp08.00.50.08"_s;
```

Comment:
I don't believe we support extended codec parameter. so `"video/webm; codec=vp08"` is probably more appropriate

Reply:
I can see `isVP8CodecConfigurationRecordSupported` which checks profile, chrome and bit depth.
In practice, this does not really matter given VP8 is SW really.